### DISCUSSION TITLE
How to know when Sse BroadcastStream is closed?

### DISCUSSION THREAD

**Discussion by azhang** [tokio-rs/axum#2749] [Q&A]
### Summary

I'm reading through https://github.com/tokio-rs/axum/discussions/1060 and https://github.com/tokio-rs/axum/discussions/2264, but they seem different since #1060  uses a plain stream instead of a broadcast stream and #2264 prepends a message instead of handling afterwards.

I am trying to add some cleanup code after each client disconnects. I've tried using guards but wasn't able to figure out when it comes to BroadcastStream. Any ideas? Any pointers much appreciated!

### axum version

0.7.5

**Comment by mladedav** [comment]
Using the drop guard pretty much the same way as in the first example should work. You just need to have something which implements `Drop` that is owned by the `impl Stream` you return from the handler. Then when the stream is dropped that object will be dropped and your code will run.

You might want to share what you've tried if this approach doesn't work for you.
